client/state/themes/upload-theme/selectors.js

```diff
@@ -22,7 +22,10 @@ export function isUploadInProgress( state, siteId ) {
  * @return {Boolean} -- True if upload has completed
  */
 export function isUploadComplete( state, siteId ) {
-	return !! get( state.themes.uploadTheme.uploadedThemeId, siteId, false );
+	return !! (
+		( ! isUploadInProgress( state, siteId ) ) &&
```

Comment:
With a site-transfer-with-theme, we may receive the theme id before the transfer is complete, so we cannot take the presence of theme id alone as an indication that a transfer has been completed.